src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/PropertyModel.cs

```diff
@@ -23,70 +23,45 @@ public class PropertyModel : ICommonModel, IBindingModel
         public PropertyModel(
             PropertyInfo propertyInfo,
             IReadOnlyList<object> attributes)
+            : base(propertyInfo?.PropertyType, attributes)
         {
-            if (propertyInfo == null)
-            {
-                throw new ArgumentNullException(nameof(propertyInfo));
-            }
-
-            if (attributes == null)
-            {
-                throw new ArgumentNullException(nameof(attributes));
-            }
-
-            PropertyInfo = propertyInfo;
-            Properties = new Dictionary<object, object>();
-            Attributes = new List<object>(attributes);
+            PropertyInfo = propertyInfo ?? throw new ArgumentNullException(nameof(propertyInfo));
         }
 
         /// <summary>
         /// Creates a new instance of <see cref="PropertyModel"/> from a given <see cref="PropertyModel"/>.
         /// </summary>
         /// <param name="other">The <see cref="PropertyModel"/> which needs to be copied.</param>
         public PropertyModel(PropertyModel other)
+            : base(other)
         {
             if (other == null)
             {
                 throw new ArgumentNullException(nameof(other));
             }
 
             Controller = other.Controller;
-            Attributes = new List<object>(other.Attributes);
             BindingInfo = BindingInfo == null ? null : new BindingInfo(other.BindingInfo);
             PropertyInfo = other.PropertyInfo;
-            PropertyName = other.PropertyName;
-            Properties = new Dictionary<object, object>(other.Properties);
         }
 
         /// <summary>
         /// Gets or sets the <see cref="ControllerModel"/> this <see cref="PropertyModel"/> is associated with.
         /// </summary>
         public ControllerModel Controller { get; set; }
 
-        /// <summary>
-        /// Gets any attributes which are annotated on the property.
-        /// </summary>
-        public IReadOnlyList<object> Attributes { get; }
-
-        public IDictionary<object, object> Properties { get; }
-
         MemberInfo ICommonModel.MemberInfo => PropertyInfo;
 
-        string ICommonModel.Name => PropertyName;
+        public new IDictionary<object, object> Properties => base.Properties;
```

Comment:
Need to do this to workaround ApiCheck. The interface contract gets satisfied by the base type which causes the property to become `sealed` (IsFinal = true). There really isn't a very good way to fix this in ApiCheck, but doing this resolves the issue.